src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,298 @@
+package duke;
+
+import duke.tasks.*;
+
+import java.util.Scanner;
+import java.util.ArrayList;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+public class Duke {
+    public static ArrayList<Task> tasks = new ArrayList<>();
+
+    public static void dividerLine() {
+        System.out.println("\t________________________________________\n");
+    }
+
+    public static void loadData() {
+        try {
+            Path dataFilePath = Paths.get("data/");
+            Files.createDirectories(dataFilePath);
+            File dataFile = new File("data/duke.txt");
+            Scanner fileReader = new Scanner(dataFile);
+            while (fileReader.hasNextLine()) {
+                String line = fileReader.nextLine();
+                String[] splitLine = line.split("\\s\\|\\s");
+                if (line.startsWith("T")) {
+                    String todo = splitLine[2];
+                    Todo t = new Todo(todo);
+                    tasks.add(t);
+                    if (splitLine[1].equals("1")) {
+                        t.markAsDone();
+                    }
+                } else if (line.startsWith("E")) {
+                    String description = splitLine[2];
+                    String date = splitLine[3];
+                    Event t = new Event(description, date);
+                    tasks.add(t);
+                    if (splitLine[1].equals("1")) {
+                        t.markAsDone();
+                    }
+                } else if (line.startsWith("D")) {
+                    String description = splitLine[2];
+                    String date = splitLine[3];
+                    Deadline t = new Deadline(description, date);
+                    tasks.add(t);
+                    if (splitLine[1].equals("1")) {
+                        t.markAsDone();
+                    }
+                }
+            }
+            fileReader.close();
+        } catch (FileNotFoundException e) {
+            ;
+        } catch (IOException e) {
+            System.out.println("\tAn error occured while trying to create a data directory :(");
+            //System.out.println(e.getMessage());
+        }
+    }
+
+    public static boolean saveData() {
+        try {
+            System.out.println("\tSaving your data...");
+            FileWriter fileWriter = new FileWriter("data/duke.txt");
+            for (int i=0; i<tasks.size(); i++) {
```

Comment:
Perhaps keep consistent spacing between operands and operations, like `i = 0` instead of `i=0`?